dlfcn_freebsd.go

```diff
@@ -5,9 +5,10 @@ package purego
 
 // Constants as defined in https://github.com/freebsd/freebsd-src/blob/main/include/dlfcn.h
 const (
-	RTLD_DEFAULT = 0xffffffff_fffffffd // Pseudo-handle for dlsym so search for any loaded symbol
-	RTLD_LAZY    = 0x00000001          // Relocations are performed at an implementation-dependent time.
-	RTLD_NOW     = 0x00000002          // Relocations are performed when the object is loaded.
-	RTLD_LOCAL   = 0x00000000          // All symbols are not made available for relocation processing by other modules.
-	RTLD_GLOBAL  = 0x00000100          // All symbols are available for relocation processing of other modules.
+	intSize      = 32 << (^uint(0) >> 63) // 32 or 64
+	RTLD_DEFAULT = 1<<intSize - 3         // Pseudo-handle for dlsym so search for any loaded symbol
```

Comment:
`1<<intSize - 2`?

Reply:
3 is correct. One lower than before because the first minus sets the bits to all 1s then -2 to clear the lower to bits

https://go.dev/play/p/7qQCqEPtFFz

Reply:
Perhaps should be written as `(1<<intSize - 1) -2`?

Reply:
https://github.com/freebsd/freebsd-src/blob/main/include/dlfcn.h#L64

Isn't `((void *) -2)` `0xfffffffe` (in 32bit)?

Reply:
You are correct. I guess the constant was wrong before then?

Reply:
Yes, unfortunately...